core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1746,21 +1751,26 @@ class KafkaApis(val requestChannel: RequestChannel,
             result.setConfigs(List.empty.asJava)
               .setNumPartitions(-1)
               .setReplicationFactor(-1)
-              .setTopicConfigErrorCode(0.toShort)
+              .setTopicConfigErrorCode(Errors.NONE.code)
           }
         }
         sendResponseCallback(results)
       }
-      adminManager.createTopics(createTopicsRequest.data.timeoutMs,
-          createTopicsRequest.data.validateOnly,
-          toCreate,
-          authorizedForDescribeConfigs,
-          handleCreateTopicsResults)
+      adminManager.createTopics(
+        createTopicsRequest.data.timeoutMs,
+        createTopicsRequest.data.validateOnly,
+        toCreate,
+        authorizedForDescribeConfigs,
+        controllerMutationQuota,
+        handleCreateTopicsResults)
     }
   }
 
   def handleCreatePartitionsRequest(request: RequestChannel.Request): Unit = {
     val createPartitionsRequest = request.body[CreatePartitionsRequest]
+    // Since version 3 of the API, the quota is strictly enforced. Any partition creation
+    // above the quota is not allowed and rejected with a THROTTLING_QUOTA_EXCEEDED error.
+    val controllerMutationQuota = quotas.controllerMutation.newQuotaFor(request, 3)
```

Comment:
As before, use named argument for `strictSinceVersion` and comment not necessary here?

Reply:
Yeah, that is a good idea.